net/net_test.go

```diff
@@ -21,6 +21,185 @@ import (
 	"testing"
 )
 
+func TestParseCIDRs(t *testing.T) {
+	testCases := []struct {
+		cidrs         []string
+		errString     string
+		errorExpected bool
+	}{
+		{
+			cidrs:         []string{},
+			errString:     "should not return an error for an empty slice",
+			errorExpected: false,
+		},
+		{
+			cidrs:         []string{"10.0.0.0/8", "not-a-valid-cidr", "2000::/10"},
+			errString:     "should return error for bad cidr",
+			errorExpected: true,
+		},
+		{
+			cidrs:         []string{"10.0.0.0/8", "2000::/10"},
+			errString:     "should not return error for good  cidrs",
+			errorExpected: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		cidrs, err := ParseCIDRs(tc.cidrs)
+		if tc.errorExpected {
+			if err == nil {
+				t.Errorf("%v", tc.errString)
+			}
+		} else {
+			if err != nil {
+				t.Errorf("%v error:%v", tc.errString, err)
+			}
+
+			// validate lengthes
+			if len(cidrs) != len(tc.cidrs) {
+				t.Errorf("cidrs should be of the same lengthes %v != %v", len(cidrs), len(tc.cidrs))
+			}
+		}
+	}
+}
+func TestDualStackIPs(t *testing.T) {
+	testCases := []struct {
+		ips       []string
```

Comment:
Thoughts on adding a `name` or `desc` field to each test table and wrap with `t.Run`?

Reply:
The old tests were using this format.. I didn't want to have some using a testing approach and others using different one. Maybe later i can go and change the entire test file..